### PR TITLE
Ruby Group Capstone Project: Code Review

### PR DESCRIPTION
**Pull request by henokkhm**
This pull request adds the following features based on the requirements of the project.

### Set up project
- Add MIT License
- Add README.md
- Add .gitignore
- Setup rubocop linter for local development
- Add pry Gem for debugging
- Add rspec Gem for testing
- Set-up Rubocop GitHub Action
- Set-up RSpec GitHub Action
- Add rubocop config file

### Project boilerplate
- Add main.rb
- Print main menu when app is started
- Add skeleton for ConsoleManager class
- Add Item, Book, Game, MusicAlbum, Genre, Label, and Author classes, 
- Add all specified methods for all classes
- Add one-to-many relationships based on UML diagram 
- Add unit tests for all base classes 
- Use Date object for all date variables 

### Collections of items
- Add BooksManager, GamesManager, MusicManager, GenreManager, LabelManager and AuthorManager classes

### Console Interface
- Add options to add and list 
    - Books
    - Games
    - Music
- Allow user to add a Label to when creating a Book
- Allow user to add a Genre to when creating a MusicAlbum
- Allow user to add a Author to when creating a Game

### Preserve Data
- Add to_hash method for all base classes
- Add methods to save and load all collections from .json file
- Make saved JSON human friendly
- Restore Books' relationship to Labels when data is loaded from .json
- Restore Games' relationship to Authors when data is loaded from .json
- Restore MusicAlbums' relationship to Genre when data is loaded from .json

### SQL Schema
- Add and test CREATE TABLE statements for 
    - Items 
    - Books
    - Games
    - Music
    - Labels
    - Genre 
    - Authors

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,174 @@
...
+
+- [📗 Table of Contents](#-table-of-contents)
+- [📖 Catalog of My Things - Ruby Group Capstone ](#-catalog-of-my-things---ruby-group-capstone-)
+  - [🛠 Built With ](#-built-with-)
+    - [Tech Stack ](#tech-stack-)
+    - [Key Features ](#key-features-)
+  - [💻 Getting Started ](#-getting-started-)
+    - [Prerequisites](#prerequisites)
+    - [Setup](#setup)
+    - [Usage](#usage)
```
Comment (MahmoudBakr23) [inline README.md:16]: - [ ] I understand that sometimes it's hard to agree upon a time as a group of 3 people. However, you've done so great so far, please rearrange a team meeting to record the maximum 5-mins presentation video in order to get the approval of this great app. 🙏💯
_Don't forget to attach the link to the presentation video here in the README file._

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello TEAM 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on creating the necessary classes for a 3 people group with the correct custom methods
- Great job with UI, it's user-friendly and so easy to follow and navigate
- Well-written tests for the archiving methods in all of the classes
- Good work preserving the data after exiting the program
- The app works as expected
- Descriptive PR and README file
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @Team  , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._


- [ ] **You did a good job by recording the video presentation as a group explaining your project as requested by the previous reviewer, however, you mistakenly exceeded the maximum length of the video (5 minutes) which was mentioned by the previous reviewer and mentioned as well in project requirements.  Kindly record another video and please make sure it doesn't exceed the length of 5 minutes.**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KDlamini — APPROVED** [review]
### Hi team @henokkhm, @Bleu-Yves-Sopoude, @Eccb7 🙋‍♀️,  

Excellent work! 💯 
You have done a great job creating a working Catalog of my Things Ruby capstone project. 🎊 🥳
You have outdone yourself in this milestone, your project is impressive!!! 👏🏾🚀 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!


## Highlights 🌞

✔️  Outstanding PR title and description
✔️  Well-documented and professional README
✔️  All aggregations and associations (1-to-many relationships) are implemented
✔️  All `can_be_archived?` methods in respective classes are present
✔️  Data is preserved in JSON format on app exit and loaded when the app starts
✔️  Database schema is set up according to the UML diagram

## Video Presentation 🎥 

✔️  The recorded video is presenting the features of the project with a maximum length of 5 minutes.
✔️ The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
✔️ All the members of the project appear in the video and talk about the project.
✔️ The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**